### PR TITLE
Use correct concurrency condition to avoid incorrect interrupts

### PR DESCRIPTION
**Pull request by damccorm**
This will help avoid incorrect interrupts to in progress workflows

For more context, see https://github.com/apache/beam/pull/27671

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by damccorm** [comment]
R: @celeste-zeng

File: .github/workflows/beam_Python_ValidatesContainer_Dataflow_ARM.yml [hunk 1]
```diff
@@ -122,4 +122,4 @@ jobs:
         uses: actions/upload-artifact@v3
         with:
           name: python-code-coverage-report
-          path: "**/pytest*.xml"
\ No newline at end of file
+          path: "**/pytest*.xml"
```
Comment (celeste-zeng) [inline .github/workflows/beam_Python_ValidatesContainer_Dataflow_ARM.yml:125]: Maybe remove the white space? Not sure why there is always white space being added every time a yml file is edited, though didn't tough the last line :(

Comment (damccorm) [reply .github/workflows/beam_Python_ValidatesContainer_Dataflow_ARM.yml:125]: This is a consequence of using the GitHub editor (as opposed to local), I think it just changes some of the line endings. I'm inclined to just leave it and ignore any of these changes

**Review by celeste-zeng — APPROVED** [review]

File: .github/workflows/beam_Python_ValidatesContainer_Dataflow_ARM.yml (hunk 1 above)
Comment (celeste-zeng) [reply .github/workflows/beam_Python_ValidatesContainer_Dataflow_ARM.yml:125]: SGTM, thanks!

**Pull request merged**